airflow/providers/amazon/aws/transfers/s3_to_redshift.py

```diff
@@ -76,6 +76,7 @@ class S3ToRedshiftOperator(BaseOperator):
         "column_list",
         "copy_options",
         "redshift_conn_id",
+        "method",
```

Comment:
The validation in the constructor of `self.method` against `AVAILABLE_METHODS` will need to move to the `execute()` method of this operator. Templated values are not rendered until _right before_ the `execute()` method is called. So, if a `method`arg is a Jinja expression or an XComArg, those explicit values will be checked against `AVAILABLE_METHODS` instead of the rendered value and the task will fail.

Reply:
Fantastic catch Josh :+1:

Reply:
Thanks @josh-fell, I moved the check to the `execute()` method

Reply:
We need to complete https://github.com/apache/airflow/issues/29069 so that test suite will notify us about these issues automatically but so far no one had the time to complete it.